Firestore/Source/Local/FSTLocalDocumentsView.mm

```diff
@@ -165,19 +169,19 @@ - (nullable FSTMaybeDocument *)localDocument:(nullable FSTMaybeDocument *)docume
  * @param documents The base remote documents to apply mutations to.
  * @return The local view of the documents.
  */
-- (FSTDocumentDictionary *)localDocuments:(FSTDocumentDictionary *)documents {
-  __block FSTDocumentDictionary *result = documents;
-  [documents enumerateKeysAndObjectsUsingBlock:^(FSTDocumentKey *key, FSTDocument *remoteDocument,
-                                                 BOOL *stop) {
-    FSTMaybeDocument *mutatedDoc = [self localDocument:remoteDocument key:key];
+- (DocumentDictionary)localDocuments:(const DocumentDictionary &)documents {
+  DocumentDictionary result = documents;
+  for (auto iter = result.begin(); iter != result.end();) {
+    FSTMaybeDocument *mutatedDoc = [self localDocument:iter->second key:iter->first];
     if ([mutatedDoc isKindOfClass:[FSTDeletedDocument class]]) {
-      result = [result dictionaryByRemovingObjectForKey:key];
+      iter = result.erase(iter);
     } else if ([mutatedDoc isKindOfClass:[FSTDocument class]]) {
-      result = [result dictionaryBySettingObject:(FSTDocument *)mutatedDoc forKey:key];
+      result[iter->first] = (FSTDocument *)mutatedDoc;
```

Comment:
Is it safe to assume that the underlying data structure will remain `std::map`, which doesn't invalidate iterators on `insert` (`unordered_map` may)?